File: P01/fizzbuzz.f
c Fortran 77 fizz buzz implementation for numbers up to 100.

      program fizzbuzz

      integer i

c     For loop from 1 to 100, inclusive
      do 10 i = 1, 100
          if (mod(i, 15) .EQ. 0) then
              write(*, '(A)') 'Fizz buzz'
          elseif (mod(i, 3) .EQ. 0) then
              write(*, '(A)') 'Fizz'
          elseif (mod(i, 5) .EQ. 0) then
              write(*, '(A)') 'Buzz'
          else
              write(*, '(I0)') i
          endif
10    continue

      stop
      end

File: P08/fizzbuzz.pl
fizz(X) :- (X mod 3) =:= 0.
buzz(X) :- (X mod 5) =:= 0.

fizzbuzz_print(X) :- fizz(X), buzz(X), write('FizzBuzz'), nl, !.
fizzbuzz_print(X) :- fizz(X), write('Fizz'), nl, !.
fizzbuzz_print(X) :- buzz(X), write('Buzz'), nl, !.
fizzbuzz_print(X) :- write(X), nl, !.

fizzbuzz(0).
fizzbuzz(N) :- N2 is N - 1, fizzbuzz(N2), fizzbuzz_print(N), !.
